Scss version

mistergraphx opened the pull request:
Bonjour,

Je propose une version scss/sass du travail effectué sur TinyTypo.
Less étant de moins en moins utilisé, y compris par les frameworks lourds (bootstrap v4, Fundation, ..), je pense qu'il est judicieux de proposer aux intégrateurs un solution plus pérenne.
J'utilise scss pour ma pars depuis 2 ans maintenant, je me propose donc de maintenir cette version ou branche ...

Bonne journée

mistergraphx committed:
Scss version

Version scss pour tinytypo

mistergraphx committed:
modif fadein fade-in en sass

tetue commented:
Yeees ! Avec plaisir !

mistergraphx commented:
;-) ha ben j'm attendais pas à une réponse si rapide ^^ c'est la magie du web !
j'ai un autre projet en cours que j'ai bêtement appellé SassySpip, dont le but est de proposer un "Kit" scss basé sur la dist et les styles de base : ceci est un portage des travaux que tu as éffectué sur la structuration des classes de spip. Bref, tinytypo pourrait ou devrais être un module de ce Kit/framework spipien dans l'idée …

mistergraphx commented:
il faudrait déterminer le niveau de spécification scss que l'on souhaite :
si c'est a partir de scssPhp, Ruby Sass/compass, libSass ... dans l'idée à savoir si on veux utiliser avec le plugin scssphp de spip, on est limité a certaines fonctions du langage.

rastapopougros commented:
Je plussoie aussi, merci. :)
J'avais commencé à le faire rapidement en local et finalement j'étais passé à autre chose, pas eu le temps.

Pour ce qui est de SCSS et SPIP, moi j'utilise SCSSPHP fourni dans le plugin idoine. Du coup ça limite en version car ça ne gère pas tout. Mais bon pour la majorité des choses ça suffit.

J'utillise TinyTypo comme base l'un des mini-frameworks de base de mon échafaudage IntéGraal ici : 
http://zone.spip.org/trac/spip-zone/browser/_squelettes_/integraal/theme/css

Mais du coup pour l'instant je l'intègre tel quel en entier, sans variable sans rien, les surcharges se faisant en tant que surcharge CSS dans les feuilles ensuite.

mistergraphx commented:
Yes, c'est pour ça que je pose la question, j'avais vu passé les commit de inteGraall (j'utilise quasiement la même base en fait...), je compile avec Gulp et Node-sass/Lib Sass pour ma part, mais pour ce qui est de TinyTipo on as pas besoin non plus de sortir l'artillerie lourde, j'imagine que l'ambition n'est pas d'en faire une usine a gaz et de rester léger... Sinon autant utiliser TypePlate ou une autre Lib dédié (ce que je fais suivant les projets ...). Donc on reste sur une compat scssphp. 
Peut être améliorer la gestion des rythmes verticaux et des tailles de titre qui sont en dur actuellement, me paraitrait bien ... du coup vu que tu utilise Bourbon/neat on peut imaginer partager les mêmes variables pour ce qui est des rytmes et autres... séparer vers un dossier base tout ce qui est html tags, serait bien ptet, ... a voir et discutter

mistergraphx commented:
Et en fait ne faut il pas faire une branche scss et une less ? ce qui éviterait que le dev qui utilise l'un ou l'autre n'embarque pas la totale ...

mistergraphx committed:
theme et encodage

Ajout d'un fichier theme, reprenant les styles de la démo du site
tinytypo

Forcer l’encodage dans les fichiers quand il y’a la présence de
caractères unicodes dans les commentaires : a corriger/supprimer et
faire le lien vers le site unicodetable ….

mistergraphx committed:
validation html5

On ne met pas de role ARIA quand la definition est déjà sémantique , on
précise le role=«  » quand on utilise une balise générique <div|span\…>

tetue commented:
Donc, c'est une très bonne idée.

Question bête : est-ce que les deux sousrep, scss et less, peuvent co-exister au dépôt, sans se gêner ? Sans trop d'incidence sur les feuilles de style générées ? Si oui, j'accepte. Car non, y'a pas plus de raison de faire une branche dédiée SCSS, qu'une branche CSS (pour celleux qui n'utilisent que le CSS de Tiny Typo), ou HTML (pour celleux qui n'utilisent que le HTML)…

Par contre ton PR contient des commits qui ne le rendent pas pas acceptables en l'état : 
- commit 5c00a916cd277d223c8ebfe331e824348b895df0 : invalide le RGAA. En effet, les roles ARIA sont nécessaires.
- commit fcfcd4bd5b571868434b3d1aebcdf07152062a5c : tinytypo est une base CSS. En tant que telle, elle ne doit pas contenir de thème, variantes, déclinaisons, etc.

Peux-tu refaire ton PR sans ces commits ?

mistergraphx commented:
> Question bête : est-ce que les deux sousrep, scss et less, peuvent co-exister au dépôt, sans se gêner ? Sans trop d'incidence sur les feuilles de style générées ?

je ne pense pas, juste il va falloir se baser sur les css générée et que les deux versions soit synchro pour restituer le même résultat.

>Peux-tu refaire ton PR sans ces commits ?

Oui, biensur je revois ça, et je re-publie.

Pour les roles aria sur des tag html5 sémantiques j'ai cru lire que c'est redondant et non recommandé, cela dit toléré comme une compatibilité/support des vieux navigateurs/readers

Cela dit c'est une page de démonstration donc pas vraiment important pour moi, comme le thème c'est juste pratique pour faire un boilerplate on pourrait ne mettre que les classes vides sans déclaration.

Au passage, je ne sais pas si tu as remarqué, mais j'ai commencé documenter le code en utilisant la méthodologie kss, ça me sert à créer des documentations automatiques pour la partie front…

mistergraphx committed:
Revert "validation html5"

This reverts commit 5c00a916cd277d223c8ebfe331e824348b895df0.

mistergraphx committed:
Revert "theme et encodage"

This reverts commit fcfcd4bd5b571868434b3d1aebcdf07152062a5c.

mistergraphx committed:
Merge remote-tracking branch 'tetue/master'

mistergraphx committed:
update 1.3.4

- revert des fichiers theme, et de la page test html
- update des dernières motifs du repo
- ajout d’une global version du package pour changer plus facilement la
version des css générées

mistergraphx committed:
Ajout de la global $tinytypo-version

de préférence importer les globals avant la doc si on veux avoir le
numero de version dans les css générées

mistergraphx commented:
Bonjour,

J'ai procédé au modifications que tu m'avais demandé et mis a jour ma branche avec les derniers commits effectués.

mistergraphx committed:
correction copier collé de la doc

mistergraphx committed:
!default flag sur les globals font-*

mistergraphx committed:
suppression de la global autoprefixer

si on utilise autoprefixer pour générer les css il gère les prefix
vendeurs obsolètes obsolètes, ça n’est donc d’aucune utilité

mistergraphx committed:
suppression des conditions relatives au prefixage vendeur

mistergraphx committed:
suppression du prefixage vendeur box-sizing est supporté sans prefix

box-sizing est supporté sans prefix
http://caniuse.com/#search=box-sizing

mistergraphx committed:
correction en scss c'est fade-out() ou transparentize

http://sass-lang.com/documentation/Sass/Script/Functions.html#transparen
tize-instance_method

a noter que  $amount est pas en % mais de 0 à 1

methbkts commented:
Bonjour,

Tiny Typo est-il toujours en développement ? Si oui, quid de la version SCSS ?

rastapopougros commented:
@methbkts pour SCSS il y a un fork là, un peu configurable : https://github.com/tcharlss/minitinytypo
Utilisé en prod sur pas mal de sites comme socle de départ.
